doc/api/url.md

```diff
@@ -250,7 +250,478 @@ properties of URL objects:
+
+The `toString()` method on the `URL` object returns the serialized URL. When
+`unicode` is `false`, the value returned is equivalent to that of `url.href`.
+
+### Class: URLSearchParams
+
+The `URLSearchParams` object provides read and write access to the querystring
+of a `URL`.
+
+```js
+const myURL = new URL('http://example.org/?abc=123');
+console.log(myURL.searchParams.get('abc'));
+  // Prints 123
+
+myURL.searchParams.append('abc', 'xyz');
+console.log(myURL.href);
+  // Prints http://example.org/?abc=123&abc=xyz
+
+myURL.searchParams.delete('abc');
+myURL.searchParams.set('a', 'b');
+console.log(myURL.href);
+  // Prints http://example.org/?a=b
+```
+
+#### Constructor: new URLSearchParams([init])
+
+* `init` {String} The URL querystring
+
+#### urlSearchParams.append(name, value)
+
+* `name` {String}
+* `value` {String}
+
+Append a new name-value pair to the query string.
+
+#### urlSearchParams.delete(name)
+
+* `name` {String}
+
+Remove all name-value pairs whose name is `name`.
+
+#### urlSearchParams.entries()
+
+* Returns: {Iterator}
+
+Returns an ES6 Iterator over each of the name-value pairs in the query string.
+Each item of the iterator is a JavaScript Array. The first item of the Array
+is the `name`, the second item of the Array is the `value`.
+
+Alias for `urlSearchParams[@@iterator]()`.
+
+#### urlSearchParams.get(name)
+
+* `name` {String}
+* Returns: {String} or `null` if there is no name-value pair with the given
+  `name`.
+
+Returns the value of the first name-value pair whose name is `name`.
+
+#### urlSearchParams.getAll(name)
+
+* `name` {String}
+* Returns: {Array}
+
+Returns the values of all name-value pairs whose name is `name`.
+
+#### urlSearchParams.has(name)
+
+* `name` {String}
+* Returns: `true` or `false`
+
+Returns `true` if there is at least one name-value pair whose name is `name`.
+
+#### urlSearchParams.keys()
+
+* Returns: {Iterator}
+
+Returns an ES6 Iterator over the names of each name-value pair.
+
+#### urlSearchParams.set(name, value)
+
+* `name` {String}
+* `value` {String}
+
+Remove any existing name-value pairs whose name is `name` and append a new
+name-value pair.
+
+#### urlSearchParams.values()
+
+* Returns: {Iterator}
+
+Returns an ES6 Iterator over the values of each name-value pair.
+
+#### urlSearchParams[@@iterator]()
+
+* Returns: {Iterator}
+
+Returns an ES6 Iterator over each of the name-value pairs in the query string.
+Each item of the iterator is a JavaScript Array. The first item of the Array
+is the `name`, the second item of the Array is the `value`.
+
+Alias for `urlSearchParams.entries()`.
+
+### require('url').originFor(url)
+
+* `url` {String | URL}
+* Returns: {Object}
+
+Returns an object representing the origin of the given URL. The origin object
+is considered to be opaque. That is, while there are properties and methods
+exported by the object, they are not considered to be part of the "public"
+API of the object.
```

Comment:
This is not a comment about this PR specifically, but what is the use of this returned object for any Node.js user, if the structure of the returned object is not part of the public API?